### DISCUSSION TITLE
KDE Plasma Wayland 下在 Flatpak 应用中使用的 Fcitx5 界面错位

### DISCUSSION THREAD

**Discussion by ImBearChild** [fcitx/fcitx5#556] [Q&A] [answered]
这个标题由于字数所限，没表述的很清楚。我使用的 Fcitx5 是由发行版提供的 RPM 包，常规程序（非 Flatpak ）一切正常，而 Flatpak 有两种情况。

1. 如果 Flatpak 内的应用使用 Wayland，那么无论是经典用户界面还是 KDE 输入面板都会错位，且无论 GTK 还是 QT 都一样。这种情况下，经典用户界面会使用 XWayland 渲染（用 xeyes 测试过了），看起来更加模糊（我采用了120%的缩放）。但理论上无论是 Flatpak 内外，都是 Wayland 环境，为什么会使 Fcitx5 使用 XWayland 呢？
 ![Classic-Wayland](https://user-images.githubusercontent.com/29907700/181878104-1415e684-897b-42e7-93e0-57bd2191aa01.png)
 ![KIM-XWayland](https://user-images.githubusercontent.com/29907700/181878511-d5f34df6-de44-4c7e-9237-de98486bc8b3.png)
2. 如果 Flatpak 内的应用使用 Xorg，那么 KDE 输入面板都会错位，但经典用户界面不会错位。
 ![Classic-XWayland](https://user-images.githubusercontent.com/29907700/181878688-a9f1ea28-4e68-45ed-a7e4-8e57a5abd5f1.png)
 ![KIM-XWayland](https://user-images.githubusercontent.com/29907700/181878634-40033d0f-6ac4-48a5-9f5c-28f962316836.png)

这种 Flatpak + Wayland 的错位的情况有没有办法解决呢？我目前找到的，有关 Fcitx5 错位的资料都表示 Gnome 配合 KIM 面板插件可以解决定位漂移问题，这在 KDE 下面使用 KIM 面板好像不太行的通。同时我没找到资料提到 KDE / Wayland / Fcitx5 / Flatpak 这四这组合下的情况。

下面附上我环境的相关信息以供参考：
操作系统： openSUSE MicroOS 20220725
KDE Plasma 版本： 5.25.3
KDE 程序框架版本： 5.96.0
Qt 版本： 5.15.5
内核版本： 5.18.11-1-default (64-位)
图形平台： Wayland
处理器： 8 × Intel® Core™ i5-9400H CPU @ 2.50GHz
内存： 15.2 GiB 内存
图形处理器： Mesa Intel® UHD Graphics 630
Fcitx5 版本：5.0.18

**Comment by wengxt** [comment, marked as answer]
当前21.08的runtime还是fcitx4的im module，已经提交了一个mr更新为fcitx5的

**Reply by CicadaSeventeen** [reply]
最好使用flathub,可以避免这个问题。

**Reply by wengxt** [reply]
看起来是 firefox 的 runtime 里面 fcitx im module 还是 fcitx4 的……我去研究一下怎么回事……我应该挺早就 pr 了用 fcitx 5 的

**Reply by wengxt** [reply]
原来是只在最新的 22.08 branch 才有，可能我应该去提交一个 pr

**Reply by wengxt** [reply]
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/9272

**Reply by wengxt** [reply]
合并了，下次runtime更新应该就好了

**Reply by ImBearChild** [reply]
> 合并了，下次runtime更新应该就好了

谢谢您的修复
